tests/cmdline/commands/test_storage.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for `verdi storage`."""
+import pytest
+
+from aiida.cmdline.commands import cmd_storage
+from aiida.common import exceptions
+
+
+@pytest.mark.usefixtures('clear_database_before_test')
+def tests_storage_info(aiida_localhost, run_cli_command):
+    """Test the ``verdi storage info`` command with the ``-statistics`` option."""
+    from aiida import orm
+    node = orm.Dict().store()
+
+    result = run_cli_command(cmd_storage.storage_info, options=['--statistics'])
+
+    assert aiida_localhost.label in result.output
+    assert node.node_type in result.output
+
+
+def tests_storage_migrate_force(run_cli_command):
+    """Test the ``verdi storage migrate`` command (with force option)."""
+    result = run_cli_command(cmd_storage.backend_migrate, options=['--force'])
+    assert result.output == ''
+
+
+def tests_storage_migrate_interactive(run_cli_command):
+    """Test the ``verdi storage migrate`` command (with interactive prompt)."""
+    from aiida.manage.manager import get_manager
+    profile = get_manager().get_profile()
+
+    result = run_cli_command(cmd_storage.backend_migrate, user_input='MIGRATE NOW')
+
+    assert 'warning' in result.output.casefold()
+    assert profile.name in result.output
+    assert 'migrate now' in result.output.casefold()
+    assert 'migration completed' in result.output.casefold()
+
+
+def tests_storage_migrate_running_daemon(run_cli_command, monkeypatch):
+    """Test that ``verdi storage migrate`` raises if the daemon is running."""
+    from aiida.engine.daemon.client import DaemonClient
+
+    def return_true():
+        return True
+
+    with monkeypatch.context() as mock_context:
+        mock_context.setattr(DaemonClient, 'is_daemon_running', return_true)
+        result = run_cli_command(cmd_storage.backend_migrate, raises=True)
```

Comment:
```suggestion
    monkeypatch.setattr(DaemonClient, 'is_daemon_running', lambda: True)
    result = run_cli_command(cmd_storage.backend_migrate, raises=True)
```

Reply:
I agree with the `lambda` shortcut, but I think I would still prefer to use the context in general as to be sure I keep the changes "under control". I mean, sure it probably just auto-resets when test function ends and if that is the only procedure in the test it might be not strictly necessary, but I still think is good practice and may help people not familiar with the monkeypatching to understand and extend the test.

Reply:
Mmm ok, I have changed if for consistency now, but I am still not convinced this is the best way to do this. I mean, it is true that their doc showcase it without the context and they only introduce the context version for a specific case, but I don't think they explicitly say the other is preferred in general. I just don't see any advantage on the simple statement while I think the context grants increased control in a situation where control is quite useful (ie, when intentionally modifying your classes so that they'll break).